fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -108,73 +126,88 @@ public ChunkReader() {
-                    txn.begin();
-                    FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
-                    List<Resource> resources2 = fhirPersistence.search(persistenceContext, resourceType).getResource();
-                    txn.commit();
-                    compartmentPageNum++;
-
-                    for (Resource res2 : resources2) {
-                        if (res2 == null) {
-                            continue;
-                        }
-                        try {
-                            FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
-                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
-                            resSubTotal++;
-                        } catch (FHIRGeneratorException e) {
-                            if (res.getId() != null) {
-                                logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
-                                        + res.getId() + "'", e);
-                            } else {
-                                logger.log(Level.WARNING,
-                                        "fillChunkDataBuffer: Error while writing resources with unknown id", e);
+
+                for (String patientId : patientIds) {
+
+                    searchContext = SearchUtil.parseQueryParameters("Patient", patientId,
+                            ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, true);
+                    do {
+                        searchContext.setPageSize(pageSize);
+                        searchContext.setPageNumber(compartmentPageNum);
+                        FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                        txn.enroll();;
+                        FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                        List<Resource> resources = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                        txn.unenroll();;
+                        compartmentPageNum++;
+
+                        for (Resource res : resources) {
+                            if (res == null || (isDoDuplicationCheck && loadedResourceIds.contains(res.getId()))) {
+                                continue;
+                            }
+                            try {
+                                FHIRGenerator.generator(Format.JSON).generate(res, chunkData.getBufferStream());
+                                chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
+                                resSubTotal++;
+                                if (isDoDuplicationCheck) {
```

Comment:
is there some documentation on this behavior?  Maybe add a comment here, just to explain when another developer sees this. (maybe on line 177) 
you can combine this logic with line 177-178

Reply:
please ignore